packages/itmat-interface/src/graphql/typeDefs.ts

```diff
@@ -104,25 +104,24 @@ type Field {
     dateDeleted: String
 }
 
+input FileMetaData {
+    deviceId: String!
+    startDate: String!
+    endDate: String!
+    postFix: String!
+}
+
 input DataClip {
     fieldId: String!
-    value: String!
+    value: String
     subjectId: String!
     visitId: String!
+    fileInput: Upload
+    metadata: FileMetaData
 }
```

Comment:
Extends the input format of clinical data uploading. 
Either `value` or (`fileInput` && `metadata`) should not be empty.
The existing scripts do not need to be updated for uploading clinical data only.

@fguitton Do you have any comments on this design?

Reply:
I think we have to be a lot more general with the metadata. Files won't always have metadata and these will also eventually integrate the organisation/user relationship as well. There might also be a great deal of annotations that could be needed in projects like Premiere or Unicorn. Perhaps also renaming `fileInput` into `file` would be more consistent. Presumably you want to conditionally render `value` and `file` obligatory based on the `dataType` of the field referenced by `fieldId`, is that correct ?

Reply:
> I think we have to be a lot more general with the metadata. Files won't always have metadata and these will also eventually integrate the organisation/user relationship as well. There might also be a great deal of annotations that could be needed in projects like Premiere or Unicorn. Perhaps also renaming `fileInput` into `file` would be more consistent. Presumably you want to conditionally render `value` and `file` obligatory based on the `dataType` of the field referenced by `fieldId`, is that correct ?

Yes, the backend check the data type and find the input from `file` (of file type) or `value` (of other types).